### ISSUE TITLE
Keep emtpy lines when senteces are between BEGIN-COMMIT

### ISSUE THREAD

**Issue by fpuga** [darold/pgFormatter#136]
When the sentences are "wrapped" between a `BEGIN;` and a `COMMIT;` all empty lines are removed:
 
Example input:
```
cat <<'EOF' | pg_format
SELECT 1;

SELECT 2;
EOF
```

Output:
```
SELECT
    1;

SELECT
    2;
```

Example input:

```
cat <<'EOF' | pg_format
BEGIN;

SELECT 1;

SELECT 2;

COMMIT;
EOF
```

Output with empty lines removed:

```
BEGIN;
SELECT
    1;
SELECT
    2
COMMIT;
```

Maybe these issues are related to this one: #70 #98 #88

**Comment by darold** [comment]
Yes, this is the intended behavior, pgFormatter remove any extra new lines between BEGIN and a COMMIT. The reason is because it is considered as a group of statements, the first formatting you mention is used for individual statements.

**Comment by fpuga** [comment]
Ok. Thanks for taking a look at this. Feel free to close.

I don`t like it but i perfectly understand your motivations about this and #138. Just to explain my case, I also use [sqitch](https://sqitch.org/) and scripts are wrapped in transactions. I commonly do more than one thing per script, for example creating a table and an associated trigger. Formatting the whole file with pgFormatter makes it no very readable (from my point of view).

But still i can individually format the sentences of the script so i can handle it.

Thanks for this tool. I test a couple of sql formatters and i understand how difficult is to write this kind of tool (and have everyone happy). pgFormatter is the best one for my use cases.

**Comment by darold** [comment]
As I said I'm not against this alternate formatting, adding a new line in grouped transaction statements is simple. This is not the case to keep original newline formatting.
  
So as I'm currently working exclusively on pgFormatting to publish a new release I think you will appreciate pgFormatter a little more: commit 8766c27 adds a new options to control this behavior :-)
```
    Add option -g, --no-grouping and CGI corresponding checkbox. By default
    pgFormatter groups all statements when they are in a transaction:
    
            BEGIN;
            INSERT INTO foo VALUES (1, 'text 1');
            INSERT INTO foo VALUES (2, 'text 2');
            ...
            COMMIT;
    
    By disabling grouping of statement pgFormatter will always add an extra
    newline characters between statements just like outside a transaction:
    
            BEGIN;
    
            INSERT INTO foo VALUES (1, 'text 1');
    
            INSERT INTO foo VALUES (2, 'text 2');
            ...
    
            COMMIT;
    
    This might add readability to not DML transactions.
```

Online demo site have been updated with the new feature.

**Comment by darold** [comment]
So it is included into release v4.1 just published.

**Comment by fpuga** [comment]
Sorry for the late answer, i already test it and it fits my needs.

Thank you.
